lib/tty.js

```diff
@@ -48,6 +48,13 @@ function WriteStream(fd) {
     writable: true
   });
 
+  // Prevents interleaved stdout/stderr output in OS X terminals.
```

Comment:
> The change is intended to fix loosing output when doing process.exit.

Technically it is intended to fix __both__.

>  Interleaving will depend on OS level buffering.

Of course, and OS X is hardcoded to 1kb so it won't change regardless.

> To clarify, IMHO the interleaving is a side effect of the blocking-ness.

Sure, but it's still a problem we need to cover.

Typically, it would just be an annoyance. You'd have all the information, just sorta mixed up.

But that's not always the case, since people often use ansi control characters in CLI apps.

Consider a problem I ran into while writing the test: At 1024kb of data, including the newline `\n`, an entire line of data no longer appears.

Why?

Somewhere down the line `\n` is being changed to `\n\r`, which overflows the buffer by one character, leaving a separate `\r` which then erases the current line. Who is going to expect that?

If not me, very few people indeed. (That was done using `Terminal.app` on OS X. Given it is the default terminal, this will be in no way isolated.)